xdsl/dialects/hw.py

```diff
@@ -845,16 +845,16 @@ def __init__(
         if visibility:
             if isinstance(visibility, str):
                 visibility = StringAttr(visibility)
-            attributes["sym_visibility"] = visibility
+            properties["sym_visibility"] = visibility
 
         return super().__init__(
-            attributes=attributes,
+            properties=properties,
             regions=[body],
         )
 
     def verify_(self) -> None:
         if self.parameters is not None:
-            # FIXME: once xDSL supports typed attributes, check that parameter
+            # FIXME: once xDSL supports typed properties, check that parameter
```

Comment:
how old is this fixme? I'm pretty sure we support typed properties and attributes...

Reply:
OK from my understanding of the issue it's actually to do with attribute the class not attribute the field of an operation, so it should still refer to attributes after the changes in this PR. Could you please revert this line? This also seems like a good thing to tackle if you're working in this space